.editorconfig: various

zowoq opened the pull request:
~~Draft until we decide to keep the `editorconfig` CI check.~~

Mic92 commented on .editorconfig:
```diff
@@ -34,3 +34,9 @@ indent_size = 2
 indent_size = 4
+
+# Disable for auto-generated hackage2nix file
+[pkgs/development/haskell-modules/hackage-packages.nix]
```
While you are at it, maybe also check if we need to add `pkgs/development/node-packages/node-packages-v*.nix`

zowoq replied:
Apart from not having final newlines they seem fine.

Some of the generated json files (`node-packages-*.json`, `recipes-archive-melpa.json`) will probably need to be excluded if we end up adding `json`.

zowoq replied:
Added `node-packages-v*.nix` and `gemset.nix`.

I guess we're probably going to end up doing this for a lot of the generated files.

zowoq commented:
@Mic92 There is about 50 `*.upstream` files scattered around the repo, are they still in use or could they be removed?

I'm going through the repo by file type and size now looking for other outliers.

Mic92 commented:
I think mainly @7c6f434c uses them. I don't know what tool he uses it for.
I personally just use https://github.com/Mic92/nix-update/ which does not need extra meta information.

zowoq commented:
> I personally just use https://github.com/Mic92/nix-update/ ...

I've been using as well. What do you think about adding it here? I send a PR it if you agree?

Mic92 commented:
> > I personally just use [Mic92/nix-update](https://github.com/Mic92/nix-update/) ...
> 
> I've been using as well. What do you think about adding it here? I send a PR it if you agree?

Feel free to add it with me as a maintainer.

7c6f434c commented:
`.upstream` files are usable with `update-walker` script which is inside Nixpkgs, and I do not always get around to updating based on them (but sometimes use them for updating).

Mic92 commented:
I think we can still merge this one.
It does not depend on the github ci check.

zowoq commented:
I guess we should still try and get all of generated files listed and add some of the more common file types.

Mic92 commented on .editorconfig:
```diff
@@ -1,36 +1,59 @@
+insert_final_newline = unset
+
+# Disable for auto-generated perl-packages file
+[pkgs/top-level/perl-packages.nix]
```
Some of the stuff might be automatic. Some of it is manual. But there might be a lot of errors still though given its size.

zowoq committed:
.editorconfig: add/disable files

add Gemfile, .json, .lock, .md, .pm

disable for some files or file types that are auto-generated
or require significant changes

zowoq commented:
I wanted to set a treewide default and require that files opt-out but there are too many miscellaneous files checked for a default to be enforced without it becoming tedious.
